docs/how-to/contribute-a-builtin.md

```diff
@@ -0,0 +1,68 @@
+# How to contribute a Builtin plugin
+
+Within a [RuleSet YAML file](../../README.md#ruleset), scriptlet probes are an escape hatch for complex, programmatic assertions. They are not intended to assert against common model patterns. For example, scriptlet probe authors may assert that certain applications or relations exist. This repetitiveness can be abstracted into Builtin assertions. Refer to this sample [RuleSet with Builtin assertions](../../tests/resources/probes/ruleset/builtins.yaml) as a starting point.
+
+Builtins are probes, serving as complementary (or isolated) assertions, depending on the scope of their assertions.
+
+```yaml
+name: RuleSet - demo
+probes:
+  - name: Probe - test passing
+    type: scriptlet
+    url: file://tests/resources/probes/python/passing.py
+  - name: My builtin
+    type: builtin/my-builtin
+    with:
+      - foo: one
+      - foo: two
```

Comment:
"foo" twice makes sense but is confusing. How about:

```suggestion
      - foo: one
        bar: two
      - foo: three
        bar: four
```

We should also mention explicitly somewhere that in favor of usability, we enable vectorizing under `with` instead of duplicating the higher level `name` etc.